rdr_service/dao/study_nph_dao.py

```diff
@@ -166,12 +167,12 @@ def __init__(self):
         super(NphSiteDao, self).__init__(Site)
 
     @staticmethod
-    def _fetch_site_id(session, site_name) -> int:
+    def _fetch_site_id(session, external_id) -> int:
         query = Query(Site)
         query.session = session
-        result = query.filter(Site.name == site_name).first()
+        result = query.filter(Site.external_id == external_id).first()
         if result is None:
-            raise NotFound(f"Site is not found -- {site_name}")
+            raise NotFound(f"Site is not found -- {external_id}")
```

Comment:
Can you move this check to the API layer? We may need to lookup sites for other purposes in the system and may not want to raise an exception on no results.